samples/cpp/benchmark_app/main.cpp

```diff
@@ -472,39 +472,58 @@ int main(int argc, char* argv[]) {
             const auto input_precision = FLAGS_ip.empty() ? ov::element::undefined : getPrecision2(FLAGS_ip);
             const auto output_precision = FLAGS_op.empty() ? ov::element::undefined : getPrecision2(FLAGS_op);
 
-            for (auto& item : model->inputs()) {
-                // if precision for input set by user, then set it to app_inputs
-                const auto& name = item.get_any_name();
-                if (user_precisions_map.count(name)) {
-                    const auto precision = getPrecision2(user_precisions_map.at(name));
-                    for (auto& info : app_inputs_info) {
-                        info.at(name).type = precision;
-                    }
+            const auto& inputs = model->inputs();
+            for (int i = 0; i < inputs.size(); i++) {
+                const auto& item = inputs[i];
+                auto iop_precision = ov::element::undefined;
+                auto type_to_set = ov::element::undefined;
+                std::string name;
+                try {
+                    // Some tensors might have no names, get_friendly_name will throw exception in that case.
+                    // -iop option will not work for those tensors.
+                    name = item.get_any_name();
+                    iop_precision = getPrecision2(user_precisions_map.at(item.get_node()->get_friendly_name()));
```

Comment:
?! Why do you need this code? OV 2.0 doesn't work with operation name. Please use name variable from the line 484.

Reply:
fixed